[Migrator] Use toLowercaseWord to lowercase SetterToProperty changes

bitjammer opened the pull request:
Post-commit review follow-up:
This works better for properties that were all-caps, such as `URL`.

Thanks @harlanhaskins for the tip!

rdar://problem/32845968

bitjammer commented:
@swift-ci Please smoke test

harlanhaskins commented on test/Migrator/pre_fixit_pass.swift.expected:
```diff
@@ -11,5 +11,6 @@ New()
 
 func foo(_ a : PropertyUserInterface) {
   a.field = 1
+  a.URL = 1
```
I think this should be `a.url = 1`

harlanhaskins commented on test/Migrator/property.swift.expected:
```diff
@@ -7,6 +7,7 @@ import Bar
 
 func foo(_ a : PropertyUserInterface) {
   a.field = 1
+  a.URL = 1
```
Ditto here

bitjammer replied:
Hm, yeah? I don't think that's what toLowercase is giving me...

harlanhaskins replied:
Ah! I think we're meant to use `toLowercaseInitialisms`

bitjammer committed:
[Migrator] Use toLowercaseWord to lowercase SetterToProperty changes

Post-commit review follow-up:
This works better for properties that were all-caps, such as `URL`.

Thanks @harlanhaskins for the tip!

rdar://problem/32845968

bitjammer commented:
@swift-ci Please smoke test

bitjammer commented:
@swift-ci Please smoke test OS X
